src/other/metadata.jl

```diff
@@ -0,0 +1,1072 @@
+
+colmetadata(x::DataFrameRows,
+            col::Int, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(x), col, key; style=style)
+colmetadata(x::DataFrameColumns,
+            col::Int, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(x), col, key; style=style)
+colmetadata(x::DataFrameRows,
+            col::Symbol, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(x), col, key; style=style)
+colmetadata(x::DataFrameColumns,
+            col::Symbol, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(x), col, key; style=style)
+colmetadata(dfr::DataFrameRows, col::ColumnIndex, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(dfr), col, key; style=style)
+colmetadata(dfc::DataFrameColumns, col::ColumnIndex, key::AbstractString; style::Bool=false) =
+    colmetadata(parent(dfc), col, key; style=style)
+
+function colmetadata(x::DataFrameRow,
+                     col::Int, key::AbstractString; style::Bool=false)
+    col_name = _names(x)[col]
+    df = parent(x)
+    val_meta, style_meta = colmetadata(df, col_name, key, style=true)
+    if style_meta !== :note
+        throw(KeyError("$key for column $col with :note style"))
+    end
+    return style ? (val_meta, style_meta) : val_meta
+end
+
+function colmetadata(x::SubDataFrame,
+                     col::Int, key::AbstractString; style::Bool=false)
+    col_name = _names(x)[col]
+    df = parent(x)
+    val_meta, style_meta = colmetadata(df, col_name, key, style=true)
+    if style_meta !== :note
+        throw(KeyError("$key for column $col with :note style"))
+    end
+    return style ? (val_meta, style_meta) : val_meta
+end
+
+colmetadata(x::DataFrameRow,
+            col::Symbol, key::AbstractString; style::Bool=false) =
+            colmetadata(x, Int(index(x)[col]), key, style=style)
+colmetadata(x::SubDataFrame,
+            col::Symbol, key::AbstractString; style::Bool=false) =
+            colmetadata(x, Int(index(x)[col]), key, style=style)
+colmetadata(x::DataFrameRow,
+            col::ColumnIndex, key::AbstractString; style::Bool=false) =
+            colmetadata(x, Int(index(x)[col]), key, style=style)
+colmetadata(x::SubDataFrame,
+            col::ColumnIndex, key::AbstractString; style::Bool=false) =
+            colmetadata(x, Int(index(x)[col]), key, style=style)
+
+"""
+    colmetadatakeys(df::AbstractDataFrame, [col::ColumnIndex])
+    colmetadatakeys(dfr::DataFrameRow, [col::ColumnIndex])
+    colmetadatakeys(dfc::DataFrameColumns, [col::ColumnIndex])
+    colmetadatakeys(dfr::DataFrameRows, [col::ColumnIndex])
+
+If `col` is passed return an iterator of column level metadata keys for which
+`metadata(x, col, key)` returns a metadata value.
+
+`SubDataFrame` and `DataFrameRow` expose only `:note` style metadata of their parent.
+
+If `col` is not passed return an iterator of `col => colmetadatakeys(x, col)`
+pairs for all columns that have metadata, where `col` are `Symbol`.
```

Comment:
It has to be `Symbol`, because it is part of DataAPI.jl API. And in DataAPI.jl strings are not required to be supported. See https://github.com/JuliaData/DataAPI.jl/pull/48/files#diff-84649239374b93ddff8d137d8b445cec97c96dea063d02946f0ae42ec7ee909dR395